single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultLabelProvider.java

```diff
@@ -225,11 +241,36 @@ private List<FeederInfo> buildFeederInfos(Terminal terminal, boolean insideVolta
             terminalQ = -terminalQ;
             terminalI = -terminalI;
         }
-        feederInfoList.add(new DirectionalFeederInfo(ARROW_ACTIVE, terminalP, svgParameters.getActivePowerUnit(), valueFormatter::formatPower));
-        feederInfoList.add(new DirectionalFeederInfo(ARROW_REACTIVE, terminalQ, svgParameters.getReactivePowerUnit(), valueFormatter::formatPower));
-        if (this.svgParameters.isDisplayCurrentFeederInfo()) {
-            feederInfoList.add(new DirectionalFeederInfo(ARROW_CURRENT, terminalI, svgParameters.getCurrentUnit(), valueFormatter::formatPower));
+        feederInfoList.add(new EdgeFeederInfo(ARROW_ACTIVE, true, terminalP, svgParameters.getActivePowerUnit(), valueFormatter::formatPower));
+        feederInfoList.add(new EdgeFeederInfo(ARROW_REACTIVE, true, terminalQ, svgParameters.getReactivePowerUnit(), valueFormatter::formatPower));
+        if (this.isDisplayCurrentFeederInfo()) {
+            boolean isDirectional = this.isDisplayArrowForCurrentFeederInfo();
+            feederInfoList.add(new EdgeFeederInfo(isDirectional ? ARROW_CURRENT : VALUE_CURRENT, isDirectional, terminalI, svgParameters.getCurrentUnit(), valueFormatter::formatCurrent));
         }
         return feederInfoList;
     }
+
+    public void setDisplayCurrentFeederInfo(boolean displayCurrentFeederInfo) {
+        this.displayCurrentFeederInfo = displayCurrentFeederInfo;
+    }
+
+    public boolean isDisplayCurrentFeederInfo() {
+        return this.displayCurrentFeederInfo;
+    }
```

Comment:
the 3 isXXX() methods are not needed, you should remove them and access directly to the field instead of making the internal state of the provider visible to the outside